### PR TITLE
Fix 2022 10 28

### PR DESCRIPTION
**Pull request by ChipsVII**

### CODE REVIEW

**Review by LupusVII — COMMENTED** [review]
Global issue with the tracking tag:
We do not want the tracking code for Google or Matomo to be used if there is nothing setup for that. When installing Smartgear, we should put the tag in tl_layout If the settings tells so. But in `src/Resources/public/examples/code_javascript_personnalise_1.js`, we just import that piece of code no matter what isn't it?

File: src/Backend/Component/Core/ConfigurationStep/Website.php [hunk 1]
```diff
@@ -359,8 +426,28 @@ protected function createLayouts(int $themeId, array $modules): array
...
         $script = file_get_contents(Util::getPublicOrWebDirectory().'/bundles/wemsmartgear/examples/code_javascript_personnalise_1.js');
-        $script = str_replace('{{config.googleFonts}}', "'".implode("','", $config->getSgGoogleFonts())."'", $script);
+        if (\count($config->getSgGoogleFonts()) > 0) {
+            $script = str_replace('{{config.googleFonts}}', "'".implode("','", $config->getSgGoogleFonts())."'", $script);
+        } else {
+            $script = preg_replace('/\/\/ -- GFONT(.*)\/\/ -- \/GFONT/s', '', $script);
+        }
+
         $script = str_replace('{{config.framway.path}}', $config->getSgFramwayPath(), $script);
+        switch ($config->getSgAnalytics()) {
```
Comment (ChipsVII) [inline src/Backend/Component/Core/ConfigurationStep/Website.php:436]: @LupusVII here is the code removing the unecessary pieces of code for analytics solutions (same logic applied for Google Fonts if none are used)

**Comment by ChipsVII** [comment]
> Global issue with the tracking tag: We do not want the tracking code for Google or Matomo to be used if there is nothing setup for that. When installing Smartgear, we should put the tag in tl_layout If the settings tells so. But in `src/Resources/public/examples/code_javascript_personnalise_1.js`, we just import that piece of code no matter what isn't it?

see src/Backend/Component/Core/ConfigurationStep/Website.php line 436

```php
// retrieve file content
$script = file_get_contents(Util::getPublicOrWebDirectory().'/bundles/wemsmartgear/examples/code_javascript_personnalise_1.js');

// manage Googe Fonts
if (\count($config->getSgGoogleFonts()) > 0) {
    $script = str_replace('{{config.googleFonts}}', "'".implode("','", $config->getSgGoogleFonts())."'", $script);
} else {
    $script = preg_replace('/\/\/ -- GFONT(.*)\/\/ -- \/GFONT/s', '', $script);
}

$script = str_replace('{{config.framway.path}}', $config->getSgFramwayPath(), $script);

// manage analytics solution
switch ($config->getSgAnalytics()) {
    case CoreConfig::ANALYTICS_SYSTEM_NONE:
        $script = preg_replace('/\/\/ -- GTAG(.*)\/\/ -- \/GTAG/s', '', $script);
        $script = preg_replace('/\/\/ -- MATOMO(.*)\/\/ -- \/MATOMO/s', '', $script);
    break;
    case CoreConfig::ANALYTICS_SYSTEM_GOOGLE:
        $script = str_replace('{{config.analytics.google.id}}', $config->getSgAnalyticsGoogleId(), $script);
        $script = preg_replace('/\/\/ -- MATOMO(.*)\/\/ -- \/MATOMO/s', '', $script);
    break;
    case CoreConfig::ANALYTICS_SYSTEM_MATOMO:
        $script = str_replace('{{config.analytics.matomo.host}}', $config->getSgAnalyticsMatomoHost(), $script);
        $script = str_replace('{{config.analytics.matomo.id}}', $config->getSgAnalyticsMatomoId(), $script);
        $script = preg_replace('/\/\/ -- GTAG(.*)\/\/ -- \/GTAG/s', '', $script);
    break;
}
```

**Comment by LupusVII** [comment]
> 
> 
> 
> > Global issue with the tracking tag: We do not want the tracking code for Google or Matomo to be used if there is nothing setup for that. When installing Smartgear, we should put the tag in tl_layout If the settings tells so. But in `src/Resources/public/examples/code_javascript_personnalise_1.js`, we just import that piece of code no matter what isn't it?
> 
> see src/Backend/Component/Core/ConfigurationStep/Website.php line 436
> 
> ```php
> // retrieve file content
> $script = file_get_contents(Util::getPublicOrWebDirectory().'/bundles/wemsmartgear/examples/code_javascript_personnalise_1.js');
> 
> // manage Googe Fonts
> if (\count($config->getSgGoogleFonts()) > 0) {
>     $script = str_replace('{{config.googleFonts}}', "'".implode("','", $config->getSgGoogleFonts())."'", $script);
> } else {
>     $script = preg_replace('/\/\/ -- GFONT(.*)\/\/ -- \/GFONT/s', '', $script);
> }
> 
> $script = str_replace('{{config.framway.path}}', $config->getSgFramwayPath(), $script);
> 
> // manage analytics solution
> switch ($config->getSgAnalytics()) {
>     case CoreConfig::ANALYTICS_SYSTEM_NONE:
>         $script = preg_replace('/\/\/ -- GTAG(.*)\/\/ -- \/GTAG/s', '', $script);
>         $script = preg_replace('/\/\/ -- MATOMO(.*)\/\/ -- \/MATOMO/s', '', $script);
>     break;
>     case CoreConfig::ANALYTICS_SYSTEM_GOOGLE:
>         $script = str_replace('{{config.analytics.google.id}}', $config->getSgAnalyticsGoogleId(), $script);
>         $script = preg_replace('/\/\/ -- MATOMO(.*)\/\/ -- \/MATOMO/s', '', $script);
>     break;
>     case CoreConfig::ANALYTICS_SYSTEM_MATOMO:
>         $script = str_replace('{{config.analytics.matomo.host}}', $config->getSgAnalyticsMatomoHost(), $script);
>         $script = str_replace('{{config.analytics.matomo.id}}', $config->getSgAnalyticsMatomoId(), $script);
>         $script = preg_replace('/\/\/ -- GTAG(.*)\/\/ -- \/GTAG/s', '', $script);
>     break;
> }
> ```

All right then! You can proceed.

**Pull request merged**